File: helixtrajectoryj/src/main/java/org/team2363/helixtrajectory/HolonomicTrajectorySample.java
package org.team2363.helixtrajectory;

public class HolonomicTrajectorySample extends TrajectorySample {

    public final double velocityX;
    public final double velocityY;
    public final double angularVelocity;

    public HolonomicTrajectorySample(double x, double y, double heading,
            double velocityX, double velocityY, double angularVelocity) {
        super(x, y, heading);

        this.velocityX = velocityX;
        this.velocityY = velocityY;
        this.angularVelocity = angularVelocity;
    }
}

File: helixtrajectorycpp/src/cpp/HolonomicTrajectorySample.cpp
#include "HolonomicTrajectorySample.h"

#include <iostream>

#include "TrajectorySample.h"

namespace helixtrajectory {

    HolonomicTrajectorySample::HolonomicTrajectorySample(double intervalDuration,
            double x, double y, double heading,
            double velocityX, double velocityY, double angularVelocity) 
            : TrajectorySample(intervalDuration, x, y, heading),
            velocityX(velocityX), velocityY(velocityY), angularVelocity(angularVelocity) {
    }

    HolonomicTrajectorySample::~HolonomicTrajectorySample() {
    }

    std::ostream& operator<<(std::ostream& stream, const HolonomicTrajectorySample& sample) {
        return stream << "{\"x\": " << sample.x
                << ", \"y\": " << sample.y
                << ", \"heading\": " << sample.heading
                << ", \"velocity_x\": " << sample.velocityX
                << ", \"velocity_y\": " << sample.velocityY
                << ", \"angular_velocity\": " << sample.angularVelocity
                << "}";
    }
}